compile/compile.go

```diff
@@ -479,7 +484,8 @@ func (c *compiler) makeClosure(code *py.Code, args uint32, child *compiler, qual
 		if reftype == symtable.ScopeCell {
 			arg = c.FindId(name, c.Code.Cellvars)
 		} else { /* (reftype == FREE) */
-			arg = c.FindId(name, c.Code.Freevars)
+			// using CellAndFreeVars in closures requires skipping Cellvars
+			arg = len(c.Code.Cellvars) + c.FindId(name, c.Code.Freevars)
```

Comment:
I couldn't find out how to use zero based subscripts to access the corresponding elements in `Code.Cellvars` and `Code.Freevars` in `Frame.Localsplus` without debugging Cpython, so I used this method to directly access `Code.CellAndFreeVars`

Reply:
Today, I dynamically debugged CPython3.4.9, and this modification is correct. For the subscripts of `FreeVars`, the length of `CellVars` needs to be added.

By the way, the complete repair of the decorator will be submitted after this PR is merged, as some modifications to the grammar are needed